crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -375,7 +375,9 @@ impl DecisionTree {
         }
         let mut modified = false;
         super::optimizations::cse_block(ctx, left, &mut merged_ins, &mut modified)?;
-
+        if modified {
+            super::optimizations::cse_block(ctx, left, &mut merged_ins, &mut modified)?;
```

Comment:
Can you elaborate on why we need to cse twice here as well? Why is twice the magic number for example rather than doing a cse while modified is true?

Reply:
Synchronisation of functions may replace function calls in one branch from a call in the other branch. The first cse update the result instructions with the reference to the replaced call, which now leaves identical result instructions (from the two branches). The second cse can now optimise them.